torch/autograd/function.py

```diff
@@ -284,6 +286,11 @@ def _iter(obj):
             for o in obj:
                 for var in _iter(o):
                     yield var
+        elif isinstance(obj, dict):
+            # We only accept primitive key types, so we needn't inspect them
```

Comment:
Can you elaborate on that comment? This function is not necessarily tied to the JIT -- it lives in `torch.autograd`.